src/main/pending_query_result.cpp

```diff
@@ -55,7 +55,8 @@ PendingExecutionResult PendingQueryResult::ExecuteTaskInternal(ClientContextLock
 
 unique_ptr<QueryResult> PendingQueryResult::ExecuteInternal(ClientContextLock &lock) {
 	CheckExecutableInternal(lock);
-	while (ExecuteTaskInternal(lock) == PendingExecutionResult::RESULT_NOT_READY) {
+	// Busy wait while execution is not finished
+	while (!IsFinished(ExecuteTaskInternal(lock))) {
```

Comment:
We could consider adding a `yield` call here as a hint to the scheduler that this is a busy loop:

```cpp
std::this_thread::yield();
```

Reply:
think that makes sense!

Reply:
I thought this method would have to run in a `DUCKDB_NO_THREADS` env as well?

Reply:
I don't know if yield makes much of a difference here (especially in the one executor per thread model), so don't think it's worth the effort (even if hidden behind a `#ifndef DUCKDB_NO_THREADS`). Up to you.